torch/_meta_registrations.py

```diff
@@ -2121,6 +2121,8 @@ def _compute_reduction_shape(self, dims, keepdim):
 def device_hint(tensor) -> "str":
     if isinstance(tensor, torch._subclasses.FakeTensor):
         return tensor.fake_device.type
+    elif hasattr(tensor, "device") and hasattr(tensor.device, "type"):
+        return tensor.device.type
```

Comment:
this feels potentially bc breaking since meta tensors will now report meta rather than cuda

Reply:
@pianpwk Why do we need this? Also did we resolve the bc breaking concern?

Reply:
I think what it was is we were running cross-checks with cpu FakeTensors and real tensors, and `device_hint()` would return "cpu" on FakeTensors, but "cuda" for real tensors, and ran into meta kernels that returned different output formats for cpu/cuda.

This is now not BC-breaking (but a bit hacky), because meta tensors now return "cuda" regardless; I'm not sure what stuff, but some stuff relies on this that I don't want to break.